examples/vision/edsr.py

```diff
@@ -0,0 +1,320 @@
+
+
+"""
+## Preparing tf.Data.Dataset object
+
+As the paper suggest to use RGB input patches of size 48×48 from lowres image with the
+corresponding highres patches. We augment the training data with random horizontal flips
+and 90 rotations.
+
+But we will be using 24x24 RGB input patches for lowres and corresponding highres
+patches(96x96).
+"""
+
+ds = train_cache
+
+# Applying Cropping, rotation and flipping
+ds = ds.map(lambda lr, hr: random_crop(lr, hr, scale=4), num_parallel_calls=AUTOTUNE)
+ds = ds.map(random_rotate, num_parallel_calls=AUTOTUNE)
+ds = ds.map(flip_left_right, num_parallel_calls=AUTOTUNE)
+# Batching Data
+ds = ds.batch(16)
+# Repeating Data, so that cardinality if dataset becomes infinte
+ds = ds.repeat(None)
+# prefetching allows later images to be prepared while the current image is being processed
+ds = ds.prefetch(buffer_size=AUTOTUNE)
+
+"""
+## Let's visualize a few sample images:
+"""
+
+lowres, highres = next(iter(ds))
+
+# Hight Resolution Images
+plt.figure(figsize=(10, 10))
+for i in range(9):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(highres[i].numpy().astype("uint8"))
+    plt.title(highres[i].shape)
+    plt.axis("off")
+
+# Low Resolution Images
+plt.figure(figsize=(10, 10))
+for i in range(9):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(lowres[i].numpy().astype("uint8"))
+    plt.title(lowres[i].shape)
+    plt.axis("off")
+
+
+def PSNR(sr, hr):
+    """
+    Computes the peak signal-to-noise ratio, measures quality of image.
+    """
+    # Max value of pixel is 255
+    psnr_value = tf.image.psnr(hr, sr, max_val=255)[0]
+    return psnr_value
+
+
+"""
+## Build a model
+
+In paper authors hav trained 3 models : EDSR, MDSR and baseline ,so for this code example
+we will be training baseline model.
+
+### Comparison of 3 residual blocks
+
+Authors compared 3 residual blocks from original resnet, SRResNet and proposed. The only
+difference is removal of batch normalization layer, Since batch normalization layers
+normalize the features, they get rid of range flexibility from networks by normalizing
+the features, it is better to remove them, Furthermore, GPU memory usage is also
+sufficiently reduced since the batch normalization layers consume the same amount of
+memory as the preceding convolutional layers.
+
+<img src="https://miro.medium.com/max/1050/1*EPviXGqlGWotVtV2gqVvNg.png" width="500" /> 
+"""
+
+# Residual Block
+def ResBlock(inputs):
+    x = layers.Conv2D(64, 3, padding="same", activation="relu")(inputs)
+    x = layers.Conv2D(64, 3, padding="same")(x)
+    x = layers.Add()([inputs, x])
+    return x
+
+
+# Upsampling Block
+def Upsampling(inputs, factor=2, **kwargs):
+    x = layers.Conv2D(64 * (factor**2), 3, padding="same", **kwargs)(inputs)
+    x = tf.nn.depth_to_space(x, block_size=factor)
+    x = layers.Conv2D(64 * (factor**2), 3, padding="same", **kwargs)(x)
+    x = tf.nn.depth_to_space(x, block_size=factor)
+    return x
+
+
+def EDSR_MODEL():
```

Comment:
`EDSRModel` -- and it should take some configuration arguments instead of hardcoding the configuration in the function body.